src/index.ts

```diff
@@ -87,15 +89,26 @@ export function ssim(
 } {
   const start = new Date().getTime();
   const options = getOptions(userOptions);
-  const ssimMap = comparison(
-    toResize(toGrayScale(validateDimensions([image1, image2, options])))
-  );
 
-  return {
-    ssim_map: ssimMap,
-    mssim: mean2d(ssimMap),
-    performance: new Date().getTime() - start
-  };
+  if (options.ssim === "weber") {
+    const [pixels1, pixels2, finalOptions] = toResize(toGrayScale(validateDimensions([image1, image2, options])));
+    const ssimMap = weberSsim(pixels1, pixels2, finalOptions);
+    return {
+      ssim_map: ssimMap,
+      mssim: ssimMap.mssim,
+      performance: new Date().getTime() - start
+    }
+  } else {
```

Comment:
If you keep the same function signature than all other methods you should be able to avoid the `if` check and just leverage the comparison function instead

Reply:
This was a performance improvement to use the in-flight mean provided by weberSsim rather than cause mean2d to run at the end which added extra delay.  I can, however, make this a lot more concise so that the if statement is not required.

Reply:
> This looks great, just some minor feedback and it looks like [1 test is failing](https://app.circleci.com/pipelines/github/obartra/ssim/188/workflows/33cafc98-1efd-4f62-a0cd-74f422def307/jobs/1297)
> 
> ![image](https://user-images.githubusercontent.com/3877773/90339154-d128d980-dfa3-11ea-8d63-34ab364eb4b7.png)
> 
> As a side note, I would also remove commented out code and format it (using `yarn format-all`) but I can do that as a follow up as well.
Are you still having this issue?

Reply:
Are you still having this issue?  [I'll handle the formatting.]

Reply:
If CI is passing I'm happy 😄 looks like it's all green now